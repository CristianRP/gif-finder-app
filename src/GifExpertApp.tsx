import { useState } from 'react'

import { AddCategory } from './components/AddCategory';

export default function GifExpertApp() {

  const [ categories, setCategories ] = useState<string[]>([ 'Spiderman', 'Youngla' ]);

  return (
    <>
      {/* title */}
      <div>GifExpertApp</div>

      {/* Input */}
      <AddCategory setCategories={ setCategories } />

      {/* Gif list */}
      <ol>
        {
          categories.map(category => {
            return <li key={category}>{ category }</li>
          })
        }
      </ol>
        {/* Gif item */}
    </>
  )
}

