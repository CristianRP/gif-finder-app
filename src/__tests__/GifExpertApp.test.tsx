import { render, screen } from '@testing-library/react';
import 'jest-environment-jsdom';

import GifExpertApp from '../GifExpertApp';

describe('Tests on <GifExpertApp /> component', () => { 

  test('should match snapshot', () => { 
    const { container } = render( <GifExpertApp /> );

    expect( container ).toMatchSnapshot();
  });

  test('should render title', () => {
    render(<GifExpertApp />);

    expect( screen.findByText('GifExpertApp') ).toBeTruthy()
  });
});
