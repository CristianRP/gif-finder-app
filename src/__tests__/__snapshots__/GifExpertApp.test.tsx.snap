// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> component should match snapshot 1`] = `
<div>
  <div>
    GifExpertApp
  </div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gif"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h3>
      Spiderman
    </h3>
    <h2>
      Loading...
    </h2>
    <div
      class="card-grid"
    />
    <h3>
      Youngla
    </h3>
    <h2>
      Loading...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
